src/atom-environment.coffee

```diff
@@ -229,6 +229,10 @@ class AtomEnvironment extends Model
 
     checkPortableHomeWritable()
 
+    # Patch the `process.env` on startup to fix the problem first documented
+    # in #4126
+    process.env = @project.getEnv()
```

Comment:
Thanks for the ideas. I'll take a deeper look at this later tonight or tomorrow.